sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1777,8 +1777,31 @@ object functions {
   def instr(str: Column, substring: String): Column = StringInstr(str.expr, lit(substring).expr)
 
   /**
-   * Locate the position of the first occurrence of substr in a string column.
+   * Returns the substring from string str before count occurrences of the delimiter delim.
+   * If count is positive, everything the left of the final delimiter (counting from left) is
+   * returned. If count is negative, every to the right of the final delimiter (counting from the
+   * right) is returned. substring_index performs a case-sensitive match when searching for delim.
    *
+   * @group string_funcs
+   * @since 1.5.0
+   */
+  def substring_index(str: String, delim: String, count: Int): Column =
```

Comment:
Let's remove this version of API. @rxin actually made some clean up, and removed the `string` (the column name) version API.